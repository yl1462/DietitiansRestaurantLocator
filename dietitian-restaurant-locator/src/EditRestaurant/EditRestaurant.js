import React, { Component } from 'react';
// import config from '../config';
import PropTypes from 'prop-types'

class EditRestaurant extends Component {
  state = {
    theRestaurant: '',
    type: '',
    id: '',
    index: ''
  }

  componentDidMount() {
    const { theRestaurant, type, id } = this.props.location.state.restaurant
    const { index } = this.props.location.state
    this.setState({
      theRestaurant,
      type,
      id,
      index
    })
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  // update the selected restaurant, with the given id
  handleSubmit = (e) => {
    const editedRestaurant = {
      theRestaurant: this.state.theRestaurant,
      type: this.state.type,
      id: this.state.id
  }
    e.preventDefault()
    this.props.editRestaurant(editedRestaurant, this.state.index)
    this.props.history.push('/home')
    // fetch(`${config.API_ENDPOINT}/api/restautant/${this.state.id}`, {
    //     method:'PATCH',
    //     headers:{
    //         'Content-Type':'application/json'
    //     },
    //     body:JSON.stringify(editedRestaurant)
    // })
    // .then(() => {
    //   console.log('edit')
    //   this.props.editRestaurant(editedRestaurant, this.state.index)
    //   this.props.history.push('/home')
    // })
    // .catch(err => {
    //     this.setState({err})
    // })
}

  render() {
    const { theRestaurant, type } = this.state

    return (
      <>
        {/* submit edited restaurant, with required name, and optional type */}
        <div className="EditRestaurant">
          <form onSubmit={this.handleSubmit}>
            <input 
              type='text' 
              value={theRestaurant}  
              name='theRestaurant'
              onChange={this.handleChange}
              className="Placeholder"
              required
              // pattern="[A-Za-z0-9\~\!\@\#\$\%\^\*\(\)\_]{3}"
              min={3}
              /><br />
            <select required onChange={this.handleChange} name='type' value={type} >
              <option value='Keto'>Keto</option>
              <option value='Mediterranean'>Mediterranean</option>
              <option value='Plant-based'>Plant-based</option>
            </select>
            <br />
            <button 
              type='submit' 
              >Update</button>
          </form>
        </div>
      </>
    );
  }
}

export default EditRestaurant;

EditRestaurant.propTypes = {
  history: PropTypes.object.isRequired
}