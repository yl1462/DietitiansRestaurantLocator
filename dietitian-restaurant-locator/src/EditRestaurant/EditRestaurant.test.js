import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EditRestaurant from './EditRestaurant'

describe('renders without crashing', () => {
  //test if page loads
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const props = {
      state: {
        index: 0,
        restaurant: {
          title: "",
          description: "",
          id: 1
        }
      }
    }

    ReactDOM.render(
      <BrowserRouter>
        <EditRestaurant location={props} />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe(`EditRestaurant component`, () => {
  const props = {
    state: {
      index: 0,
      restaurant: {
        title: "",
        description: "",
        id: 1
      }
    }
  }

  //test if testing updated restaurant renders
  it('renders the new restaurant given props', () => {
    const wrapper = shallow(<EditRestaurant location={props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

