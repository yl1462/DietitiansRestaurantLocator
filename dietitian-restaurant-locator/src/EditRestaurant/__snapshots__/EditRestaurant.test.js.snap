// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRestaurant component renders the new restaurant given props 1`] = `
<Fragment>
  <div
    className="EditRestaurant"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="Placeholder"
        min={3}
        name="the_restaurant"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <br />
      <select
        name="type"
        onChange={[Function]}
        required={true}
        value=""
      >
        <option
          value="Keto"
        >
          Keto
        </option>
        <option
          value="Mediterranean"
        >
          Mediterranean
        </option>
        <option
          value="Plant-based"
        >
          Plant-based
        </option>
      </select>
      <br />
      <button
        type="submit"
      >
        Update
      </button>
    </form>
  </div>
</Fragment>
`;
